docs/content/docs/features/custom-schemas/custom-blocks.mdx

```diff
@@ -143,15 +142,120 @@ type ReactCustomBlockImplementation = {
+  }[];
+  plugins?: Plugin[];
+  tiptapExtensions?: AnyExtension[];
+}
+
+const customBlockExtensionOptions: BlockNoteExtensionOptions = {
+  key: "customBlockExtension",
+  keyboardShortcuts: ...,
+  inputRules: ...,
+  plugins: ...,
+  tiptapExtensions: ...,
+}
+
+const CustomBlock = createReactBlockSpec(
+  {
+    type: ...,
+    propSchema: ...,
+    content: ...,
+  },
+  {
+    render: ...,
+    ...
+  },
+  [createBlockNoteExtension(customBlockExtensionOptions)]
+)
+```
+
+Let's go over the options that can be passed into `createBlockNoteExtension`:
+
+`key:` The name of the extension.
+
+`keyboardShortcuts?:` Keyboard shortcuts can be used to run code when a key combination is pressed in the editor. The key names are the same as those used in the [`KeyboardEvent.key` property](https://developer.mozilla.org/en-US/docs/Web/API/UI_Events/Keyboard_event_key_values). Takes an object which maps key name combinations (e.g. `Meta+Shift+ArrowDown`) to functions, which return `true` when the key press event is handled, or `false` otherwise. The functions have a single argument:
+
+- `ctx:` An object containing the BlockNote editor instance.
+
+`inputRules?:` Input rules update blocks when given regular expressions are found in them. Takes an array of objects. Each object has a `find` field for the regular expression to find, and a `replace` field, for a function that should run on a match. The function should return a [`PartialBlock`](docs/reference/editor/manipulating-content#partial-blocks) which specifies how the block should be updated, or avoid updating it. It also has a single argument:
+
+- `props:` An object containing the result of the regular expression match, a range for the [Prosemirror position indices](https://prosemirror.net/docs/guide/#doc.indexing) spanned by the match, and the BlockNote editor instance.
+
+`plugins?:` An array of [ProseMirror plugins](https://prosemirror.net/docs/ref/#state.Plugin_System).
+
+`tiptapExtensions?:` An array of [TipTap extensions](https://tiptap.dev/docs/editor/extensions/custom-extensions/create-new).
+
+### Block Config Options
+
+In some cases, you may want to have a customizable block config. For example, you may want to be able to have a code block with syntax highlighting for either web or embedded code, or a heading block with a flexible number of heading levels. You can use the same API for this use case, with some minor changes:
+
+```typescript
+// Arbitrary options that your block can take, e.g. number of heading levels or
+// available code syntax highlight languages.
+type CustomBlockConfigOptions = {
+  ...
+}
+
+const CustomBlock = createReactBlockSpec(
+  createBlockConfig((options: CustomBlockConfigOptions) => ({
+    type: ...,
+    propSchema: ...,
+    content: ...,
+  })),
+  (options: CustomBlockConfigOptions) => ({
+    render: ...,
+    ...
+  })
+)
+```
+
 ## Adding Custom Blocks to the Editor
 
 Finally, create a BlockNoteSchema using the definition of your custom blocks:
```

Comment:
Probably worth talking about schema extending, allowing you to add more blocks without having to spread